.github/actions/templates/avm-validateModuleDeployment/action.yml

```diff
@@ -57,6 +57,43 @@ runs:
         creds: ${{ env.AZURE_CREDENTIALS }}
         enable-AzPSSession: true
 
+    # [Set Deployment Location] task(s)
+    # ---------------------------
+    - name: "Set Deployment Location"
+      id: set-deployment-location
+      uses: azure/powershell@v1
+      with:
+        azPSVersion: "latest"
+        inlineScript: |
+          # Grouping task logs
+          Write-Output '::group::Get Recommended Regions'
+
+          # Load used functions
+          . (Join-Path $env:GITHUB_WORKSPACE 'avm' 'utilities' 'pipelines' 'sharedScripts' 'Get-AzRecommendedRegions.ps1')
+
+          # Set fucntion input parameters
+          $functionInput = @{
+              excludedRegions = @("westeurope","westus2","eastus2","centralus","southcentralus","brazilsouth","koreacentral","qatarcentral")
+              usePairedRegionsOnly = $true
+              useKnownRegionsOnly = $true
+              knownRegionList = @("northeurope", "eastus", "polandcentral", "germanywestcentral", "francecentral", "westus3", "australiaeast", "uksouth")
+          }
+
+          $regions = Get-AzRecommendedRegions @functionInput
+          Write-Verbose "Recommended regions: $($regions.Count)" -Verbose
+
+          $index = Get-Random -Maximum ($regions.Count)
+          Write-Verbose "Random index: $index" -Verbose
+
+          $location = $regions[$index].Location
+          Write-Verbose "Random location: $location" -Verbose
```

Comment:
Why not have this logic in the function and use it to get a randomized location? 

The less logic we have in the pipeline the better.

Reply:
I did originally but didn't want to make a seprate function. But can do this if we want!

Reply:
Why separate? Do we need a function that only returns RecommendedRegions somewhere else? I was more thinking along the lines of having 1 function like `Get-DeploymentLocation` (just the first thing that came to mind) and just have the functionality in there.

Reply:
Adding new function

Reply:
"In the future we may want" is usually an anti-pattern 😄 Don't get me wrong, it's honorable, and I'll give in as soon as you give me one good future use case where we may want to use it and couldn't redesign 😉 because the alternative is to introduce complexity now, for no added value.

Reply:
Consildated to 1 function. please review